donation-tracker/src/utils/dataStore.ts

```diff
@@ -1,316 +1,486 @@
-		);
-		// register next execution of data update
-		this.registerDataUpdate();
-	}
+      this.hasLocalData = true;
+      if (this.isDataOutdated()) {
+        this.debugLog('DataStore: Local data outdated, querying online source scheduled');
+        if (!this.isGatsbyBuild) setTimeout(this.refreshData.bind(this), 100);
+      } else {
+        if (!this.isGatsbyBuild) this.registerDataUpdate();
+      }
+    } else {
+      this.initWithEmptyLocalData('DataStore: No local data found, querying online source scheduled');
+    }
+  }
 
-	/**
-	 * Internal function to extract the data from the lines of the CSV file
-	 * @param dataRows The array containing the lines of the SV file, including header line
-	 * @returns An array with the DonationItems that were extracted
-	 */
-	private extractDataLines(dataRows: string[]): Array<DonationItem> {
-		this.debugLog('DataStore: Data extracting data lines from online source');
-		const tmpResult: Array<DonationItem> = [];
-		for (let idx = 1; idx < dataRows.length; idx++) {
-			const tmpLineData = this.parseCsvLine(dataRows[idx]);
-			if (tmpLineData.length >= 5) {
-				const tmpObject: DonationItem = {
-					article: tmpLineData[0].toString(),
-					campaignKey: tmpLineData[1].toString(),
-					neededOverall: Number.parseInt(tmpLineData[2].toString()),
-					alreadyDonated: Number.parseInt(tmpLineData[3].toString()),
-					remainingNeed: Number.parseInt(tmpLineData[4].toString()),
-					unit: tmpLineData[5].toString()
-				}
-				tmpResult.push(tmpObject);
-			}
-		}
-		return tmpResult;
-	}
+  /**
+   * Internal function to check if the expected data version of the code matches the data version found locally
+   * @returns true if version matches with code, false if not
+   */
+  private isLocalDataVersionOk() {
+    this.debugLog('DataStore: Data Version Check');
+    if (typeof this.localData!.version === 'undefined') {
+      // missing version field = outdated
+      return false;
+    } else {
+      if (this.localData!.version !== this.dataVersion) {
+        // version mismatch = outdated
+        return false;
+      }
+    }
+    return true;
+  }
 
-	/**
-	 * Function to check if the local data is outdated (older then the configured amount of minutes).
-	 * @returns A boolean result with the answer to the question
-	 */
-	private isDataOutdated() {
-		this.debugLog('DataStore: Checking if data is outdated...');
+  /**
+   * Internal function to end class init with no local data
+   * @param logMessage Message that should be logged as reason
+   */
+  private initWithEmptyLocalData(logMessage: string) {
+    this.debugLog(logMessage);
+    this.localData = null;
+    this.hasLocalData = false;
+    if (!this.isGatsbyBuild) setTimeout(this.refreshData.bind(this), 100);
+  }
+
+  /**
+   * Internal function to register the next data update
+   */
+  private registerDataUpdate() {
+    if (PageConfiguration.AutoRefresh) {
+      let secondsUntilRefresh = this.getSecondsUntilRefresh();
```

Comment:
```
    // use default refresh time, if it can not be calculated
    const secondsUntilRefresh = this.getSecondsUntilRefresh() ?? 1000 * 60 * this.maxDataAgeInMinutes;
```

Reply:
done